Source/WebCore/page/LocalDOMWindow.cpp

```diff
@@ -2458,39 +2460,175 @@ void LocalDOMWindow::finishedLoading()
     }
 }
 
+PerformanceEventTiming::InteractionID& LocalDOMWindow::ensureUserInteractionValue()
+{
+    // User interaction value should be initialized with a random number from 100 to 10000:
+    if (m_userInteractionValue < 100) [[unlikely]]
+        m_userInteractionValue = 100 + WTF::cryptographicallyRandomNumber<uint64_t>() % 9901;
+    return m_userInteractionValue;
+}
+
+void LocalDOMWindow::increaseInteractionCount()
+{
+    ++m_interactionCount;
+    // m_userInteractionValue should be increased by a small integer greater
+    // than 1 to "discourage developers from considering it as a counter":
+    ensureUserInteractionValue() += 7;
+};
+
 PerformanceEventTiming::Candidate LocalDOMWindow::initializeEventTimingEntry(const Event& event, EventTypeInfo typeInfo)
 {
-    // FIXME: implement InteractionId logic
     auto processingStart = performance().nowInReducedResolutionSeconds();
+    PerformanceEventTiming::InteractionID interactionId = 0;
     LOG_WITH_STREAM(PerformanceTimeline, stream << "Initializing event timing entry (type=" << event.type() << ") at t=" << processingStart);
+    // Stateful interactionId logic as described in https://www.w3.org/TR/event-timing/#sec-computing-interactionid :
+    switch (typeInfo.type()) {
```

Comment:
We should extract this switch into a separate helper function named "computeInteractionId" instead of adding a comment like this.